src/procedures/proc_functions.c

```
@@ -0,0 +1,145 @@
+    while(raxNext(&pdata->iter)){
+
+        AR_FuncDesc *func = (AR_FuncDesc*)pdata->iter.data;
+    
+        if(func->name &&  !func->internal){
+            SITypes_SignatureToString(func->name, func->ret_type, func->types, buf, bufferLen); // get the function signature into buf
+            *pdata->yield_name = SI_ConstStringVal(func->name);
+            *pdata->yield_signature = SI_DuplicateStringVal(buf);    
+            *pdata->yield_description = func->description ? SI_ConstStringVal(func->description) : SI_ConstStringVal("No description available");
+            return pdata->output;
+        }
+    }
```

cppcheck

[error] Common realloc mistake

(memleakOnRealloc)